### PR TITLE
Feature/better simple classes

### PR DESCRIPTION
**Pull request by Tinche**
This should be safe to merge in any case. Generate better test cases.

The .gitignore addition is for my convenience ;)

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -74,6 +74,14 @@ def gen_attr_names():
             yield outer + inner
 
 
+def maybe_underscore_prefix(source):
+    """A generator to sometimes prepend an underscore."""
```
Comment (hynek) [inline tests/utils.py]: *cough* `"""` on separate lines *cough*

I should really look into pylint so I don’t get to be the asshole that pesters everyone. :)

**Comment by Tinche** [comment]
There, easily fixed. I really do prefer the one-line version when the comment is short-ish and fits on one line though.

File: .gitignore [hunk 2]
```diff
@@ -6,4 +6,5 @@ docs/_build/
 htmlcov
 dist
 .cache
-.hypothesis
\ No newline at end of file
+.hypothesis
+.venv*
```
Comment (hynek) [inline .gitignore]: oh and please don't do this I don’t want put stuff there that doesn’t affect everyone.  do you know about `.git/info/exclude`?

**Comment by Tinche** [comment]
Googling around, I added the rule to the global gitignore file (~/.config/git/ignore). Problem fixed forever.

**Pull request merged**